responsibleai/responsibleai/managers/data_balance_manager.py

```diff
@@ -0,0 +1,336 @@
+                )
+            )
+
+        if self._df[self._target_column].unique().size > 2:
+            raise ValueError(
+                (
+                    f"The target_column '{self._target_column}' must contain"
+                    " at most 2 unique values (a positive label and a"
+                    " negative label) to compute data balance measures."
+                )
+            )
+
+        if self._task_type != TaskType.CLASSIFICATION:
+            raise ValueError(
+                (
+                    f"The task_type '{self._task_type}' must be"
+                    " 'classification' to compute data balance measures."
+                )
+            )
+
+    def compute(self):
+        """
+        Computes data balance measures on the dataset.
+        """
+        if not self._is_added:
+            return
+
+        self._validate()
+
+        self._df = DataBalanceHelper.prepare_df(
+            df=self._df,
+            target_column=self._target_column,
+            pos_label=self._pos_label,
+        )
+
+        feature_balance_measures = FeatureBalanceMeasures(
+            cols_of_interest=self._cols_of_interest,
+            label_col=self._target_column,
+        ).measures(dataset=self._df)
+        distribution_balance_measures = DistributionBalanceMeasures(
+            cols_of_interest=self._cols_of_interest
+        ).measures(dataset=self._df)
+        aggregate_balance_measures = AggregateBalanceMeasures(
+            cols_of_interest=self._cols_of_interest
+        ).measures(dataset=self._df)
+
+        self._set_data_balance_measures(
+            feature_balance_measures=feature_balance_measures,
+            distribution_balance_measures=distribution_balance_measures,
+            aggregate_balance_measures=aggregate_balance_measures,
+        )
+
+    def _set_data_balance_measures(
+        self,
+        feature_balance_measures: pd.DataFrame,
+        distribution_balance_measures: pd.DataFrame,
+        aggregate_balance_measures: pd.DataFrame,
+    ):
+        """
+        Take the data balance measure DataFrames, transform them into
+        a dictionary, and set them as an instance variable.
+
+        :param feature_balance_measures: Feature balance measures.
+        :type feature_balance_measures: pd.DataFrame
+        :param distribution_balance_measures: Distribution balance measures.
+        :type distribution_balance_measures: pd.DataFrame
+        :param aggregate_balance_measures: Aggregate balance measures.
+        :type aggregate_balance_measures: pd.DataFrame
+        """
+        self._data_balance_measures = (
+            DataBalanceHelper.transform_measures_to_dict(
+                feature_balance_measures=feature_balance_measures,
+                distribution_balance_measures=distribution_balance_measures,
+                aggregate_balance_measures=aggregate_balance_measures,
+            )
+        )
+
+    def get(self):
```

Comment:
why do we have both get and get_data?  I think maybe get can be removed?

Reply:
The base class for managers (`BaseManager`) defines `get()` and `RAIBaseInsights` calls `.get()` for every manager, so I think it is required for me to implement `get()`!

As for `get_data()`, since data balance manager doesn't have its own `XYZData` class to return to `RAIInsights`, I'm just calling `get()` inside it so it can be used like this:

![image](https://user-images.githubusercontent.com/64443771/175133558-96a786e1-0465-4126-af95-de4a9fce1a7b.png)

Reply:
"since data balance manager doesn't have its own XYZData class to return to RAIInsights"
Yeah I guess that's my confusion, why it doesn't have such a class.  Maybe it's ok, just seems very different from the other managers.

Reply:
My thought process was that if the measures are a part of an `IDataset` and are a simple `dict`, then they don't need their own class. However, your point that every manager has its own `XYZData` class totally makes sense. Do you think I should introduce an `XYZData` class here?